src/tpm.c

```diff
@@ -85,14 +85,233 @@ tss_esys_context(void)
 	return esys_ctx;
 }
 
+static bool
+tpm_get_tpm_property(TPM2_PT property, uint32_t *value)
+{
+	ESYS_CONTEXT *esys_ctx = tss_esys_context();
+	TPMS_CAPABILITY_DATA *cap_data = NULL;
+	TPMI_YES_NO more_data;
+	TPML_TAGGED_TPM_PROPERTY *props = NULL;
+	TSS2_RC rc;
+	bool okay = false;
+
+	if (value == NULL)
+		return false;
+
+	rc = Esys_GetCapability(esys_ctx, ESYS_TR_NONE, ESYS_TR_NONE,
+			ESYS_TR_NONE, TPM2_CAP_TPM_PROPERTIES, property,
+			1, &more_data, &cap_data);
+	if (rc != TSS2_RC_SUCCESS)
+		return tss_check_error(rc, "Esys_GetCapability (TPM_PROPERTIES) failed");
+
+	if (cap_data == NULL) {
+		error("Empty CAP data (TPM_PROPERTIES)\n");
+		return false;
+	}
+
+	if (cap_data->capability != TPM2_CAP_TPM_PROPERTIES) {
+		error("Wrong CAP data (TPM_PROPERTIES)\n");
+		goto out;
+	}
+
+	props = &cap_data->data.tpmProperties;
+	if (props->count != 1) {
+		error("Got more than 1 property\n");
+		goto out;
+	}
+
+	if (props->tpmProperty[0].property != property) {
+		error("Property not match\n");
+		goto out;
+	}
+
+	*value = props->tpmProperty[0].value;
+
+	okay = true;
+out:
+	if (cap_data)
+		free(cap_data);
+
+	return okay;
+}
+
+#ifdef TPM2_CAP_AUTH_POLICIES
```

Comment:
When is this not defined?

Reply:
It was added to tpm2-tss since 3.2.0 while openSUSE Leap 15.6 still uses 3.1.1.